app/src/toolbox.cpp

```diff
@@ -166,6 +166,48 @@ void ToolBoxWidget::updateUI()
 {
 }
 
+void ToolBoxWidget::onToolSetActive(ToolType toolType)
```

Comment:
The problem is that we can't do that currently, we basically only want to update the UI state, eg. show that the tool is checked, not call toolmanager again as it would cause a cyclic reference, since we're coming from tool manager in this case. The way I see it is that xyzOn are methods that call ui -> manager, where as the onToolSetActive is a manager -> ui.

Maybe it's not the best solution, do you have a better idea?

Reply:
Sorry, I didn’t think about that. But even so, or rather because of it, the xyzOn() methods can be simplified. I’ll send a PR.